test/e2e/utils/datainserter.go

```diff
@@ -173,6 +173,58 @@ func (di *DataInserter) Read() ([]*TestData, error) {
 	return res, nil
 }
 
+func (di *DataInserter) StartContinuousReads(ctx context.Context) (func() error, error) {
+	ctx, cancel := context.WithCancel(ctx)
+
+	stmt, names := di.table.SelectAll()
+	q := di.session.ContextQuery(ctx, stmt, names).BindStruct(&TestData{})
+
+	var wg sync.WaitGroup
+	errCh := make(chan error, 1)
+
+	trafficStarted := make(chan struct{})
+	notifyTrafficStarted := sync.OnceFunc(func() {
+		close(trafficStarted)
+	})
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		var res []*TestData
+		for {
+			err := q.Select(&res)
+			if err != nil {
+				if errors.Is(err, context.Canceled) {
+					errCh <- nil
+					return
+				}
+				errCh <- fmt.Errorf("can't select data: %w", err)
+				return
+			}
+
+			notifyTrafficStarted()
+		}
+	}()
+
+	select {
+	case <-time.After(10 * time.Second):
+		cancel()
+		return nil, fmt.Errorf("traffic not started before timeout")
```

Comment:
shouldn't this also call q.Release() and wait for the goroutine?

Reply:
yep, done